### DISCUSSION TITLE
Signing a transaction with Deeplink

### DISCUSSION THREAD

**Discussion by theParsius** [phantom/docs#82] [General]
Hello there,
I'm a part of the Honeyland team and we have some problems with signing a transaction with Deeplink
1. After signing the transaction Phantom adds empty instructions to the transaction
2. Phantom returns an empty transaction

Both issues occur on Android and iOS

**Comment by adamdelphantom** [comment]
Hey @theParsius,

Thank you for brining this over from discord!

Can you provide an example deep link url and transaction? w/ the empty instructions.

And an example deep link url that results in an empty transaction?

**Reply by theParsius** [reply]
This is the URL we use for deep link
https://phantom.app/ul/v1/signAllTransactions?" +
                          $"dapp_encryption_public_key={Base58Encoding.Encode(UserWalletManager.Instance.dappPublicKey)}" +
                          $"&nonce={Base58Encoding.Encode(nonceEncrypt)}" +
                          $"&redirect_link={UserWalletManager.RedirectLink}+?{redirectKey}=true" +
                          $"&payload={Base58Encoding.Encode(payloadEncrypt)

I can't give you an example of the transaction because the issue is not consistent.

**Reply by adamdelphantom** [reply]
@theParsius Thank you for providing that.

The next release should be out in about ~2 weeks and will have some enhanced logs available for debugging deep link issues.

It's worth trying to reproduce the issue w/ 23.12.0 and checking the logs available via "Settings > Security & Privacy."

If there's nothing helpful there, I would try again when 23.13.0 is out in approximately
2 weeks.

**Reply by theParsius** [reply]
Thank you, I will test again and check the log section and get back to you

**Reply by theParsius** [reply]
I checked the logs and I couldn't find anything useful. Our main problem is that after signing the transaction with Phantom via the deep link, there seems to be some instruction added to the transaction that we don't use. That causes many bugs within our game that uses blockchain transactions.

**Reply by adamdelphantom** [reply]
Yeah, the next release should have some deep link specific debugging logs that may be helpful.

Regarding the additional instructions:

Are they intermittently included? 

Or is it every time?

If you can't provide an example transaction, can you indicate the additional instruction?

**Reply by theParsius** [reply]
I believe it is added every time. I will attach the examples below
[DeserializedPhantomTransaction.txt](https://github.com/phantom/docs/files/12503736/DeserializedPhantomTransaction.txt)
[PhantomTransaction.txt](https://github.com/phantom/docs/files/12503737/PhantomTransaction.txt)

**Reply by adamdelphantom** [reply]
@theParsius Is this reproducible with your current production apps iOS or Android?

**Reply by theParsius** [reply]
Yes, but due to security reasons, you won't be able to see the log. You'll need a special version to see the log. If you want I can send you the special version in a private message.
